cli/cmd/main.go

```diff
@@ -785,3 +866,90 @@ func DataCommand(c *cli.Context) error {
 
 	return nil
 }
+
+// DeleteCommand runs the command for deleting data from the Viam cloud.
+func DeleteCommand(c *cli.Context) error {
+	if c.String(dataFlagDataType) != dataTypeBinary && c.String(dataFlagDataType) != dataTypeTabular {
+		return errors.Errorf("type must be binary or tabular, got %s", c.String("type"))
+	}
+
+	filter := &datapb.Filter{}
```

Comment:
Since the following logic on lines 876-933 is shared between this data and deletion commands, would suggest extracting into a helper method `createDataFilter(filter *datapb.Filter) error`

Reply:
great suggestion, done!